unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs

```diff
@@ -79,14 +130,13 @@ public void UpdateData(UserProfileModel newModel)
         model.muted = newModel.muted;
         model.version = newModel.version;
 
-        if (model.snapshots != null && faceSnapshotDirty)
+        if (faceSnapshotDirty)
             snapshotObserver.RefreshWithUri(face256SnapshotURL);
 
-        if (model.snapshots != null && bodySnapshotDirty)
+        if (bodySnapshotDirty)
             bodySnapshotObserver.RefreshWithUri(bodySnapshotURL);
 
-        if (isModelDirty)
-            OnUpdate?.Invoke(this);
+        OnUpdate?.Invoke(this);
```

Comment:
To confirm, is it safe to remove the `isModelDirty` bool? Wouldnt then the `OnUpdate` function be called even if the model has no changes?

Reply:
I think that `isModelDirty` was a small optimization. However, many clients of this class are highly dependent on this callback. Thus, not calling it can cause some systems wait it for long (possibly infinite) time. For example in the bug-report #5058 you can see QA noticed very long loading time. It was exactly because LoadingScreen or other systems were awaiting OnUpdate to happen